client/stmt.go

```diff
@@ -265,9 +265,14 @@ func (c *Conn) Prepare(query string) (*Stmt, error) {
 	s.params = int(binary.LittleEndian.Uint16(data[pos:]))
 	pos += 2
 
-	// warnings
-	s.warnings = int(binary.LittleEndian.Uint16(data[pos:]))
-	// pos += 2
+	// reserved
+	pos += 1
```

Comment:
I just wonder why the old code behaves wrong but user didn't report it 😢 Maybe they didn't check the warning counts? 

There are already some unit tests using PREPARE, can you modify one case to check `func (s *Stmt) WarningsNum() int` is correct?